### ISSUE TITLE
Project #1 Doubt : Loss becoming constant

### ISSUE THREAD

**Issue by prateek27** [vineetm/ell-881-2018-deep-learning#27]
I trained my Model for NLI task. After few epochs loss is becoming constant. I tried to run it on smaller dataset, the loss was decreasing constant but on larger dataset it first increased and then became constant. What does this mean ? In my MLP I am using one hidden layer with 256 units, followed by output layer and the following  -
'tf.reduce_mean( tf.nn.softmax_cross_entropy_with_logits_v2(logits=predicted_y, labels=desired_y) )' function for loss. 

INFO:tensorflow:Sun Oct  7 22:38:03 2018: Epoch: 1 Train_loss: 28167.1582
INFO:tensorflow:Sun Oct  7 23:11:03 2018: Epoch: 2 Train_loss: 28347.4082
INFO:tensorflow:Sun Oct  7 23:44:03 2018: Epoch: 3 Train_loss: 32780.3086
INFO:tensorflow:Mon Oct  8 00:17:25 2018: Epoch: 4 Train_loss: 32779.1523
INFO:tensorflow:Mon Oct  8 00:50:37 2018: Epoch: 5 Train_loss: 32779.1172
INFO:tensorflow:Mon Oct  8 01:23:36 2018: Epoch: 6 Train_loss: 32779.1016
INFO:tensorflow:Mon Oct  8 01:56:44 2018: Epoch: 7 Train_loss: 32779.1016
INFO:tensorflow:Mon Oct  8 02:30:03 2018: Epoch: 8 Train_loss: 32779.1016
INFO:tensorflow:Mon Oct  8 03:03:22 2018: Epoch: 9 Train_loss: 32779.1016
INFO:tensorflow:Mon Oct  8 03:36:39 2018: Epoch: 10 Train_loss: 32779.1016

**Comment by vineetm** [comment]
@prateek27 Not sure if you resolved this. There can be many reasons for this to happen. I am assuming `Train_Loss` here is for the batch executed last, or is it for the entire dataset? Further some quick checks, `predicted_y` is un-normalized logits right and not a probability distribution?

**Comment by prateek27** [comment]
@vineetm My issue is now resolved. Earlier I was computing train loss over entire dataset, but as mentioned in assignment now I am computing now loss over each in batch in just 1 epoch. I channged these two things - 
1) Taking in account sentence lengths (that was a major problem)
2) Increased buffer size for shuffling

Now the results appear something - 
INFO:tensorflow:Mon Oct  8 12:01:27 2018: Epoch 0: Batch 100 Batch_loss: 1.2418
INFO:tensorflow:Mon Oct  8 12:01:45 2018: Epoch 0: Batch 200 Batch_loss: 0.9780
INFO:tensorflow:Mon Oct  8 12:02:05 2018: Epoch 0: Batch 300 Batch_loss: 0.9020
INFO:tensorflow:Mon Oct  8 12:02:26 2018: Epoch 0: Batch 400 Batch_loss: 0.7898
INFO:tensorflow:Mon Oct  8 12:02:45 2018: Epoch 0: Batch 500 Batch_loss: 0.7918
INFO:tensorflow:Mon Oct  8 12:03:02 2018: Epoch 0: Batch 600 Batch_loss: 0.8236
INFO:tensorflow:Mon Oct  8 12:03:20 2018: Epoch 0: Batch 700 Batch_loss: 0.8161
INFO:tensorflow:Mon Oct  8 12:03:38 2018: Epoch 0: Batch 800 Batch_loss: 0.8025
INFO:tensorflow:Mon Oct  8 12:03:55 2018: Epoch 0: Batch 900 Batch_loss: 0.6879
